### PR TITLE
#166265842 sign-up validation

### PR DESCRIPTION
**Pull request by PelumiAlesh**
### What does this PR do?
Add validation to the sign-up endpoint
#### Description of Task to be completed?
When a user tries to sign up with the wrong or unsupported input format, the user should receive descriptive signup/Registration validation error(s)
#### Any background context you want to provide?
No
#### What are the relevant pivotal tracker stories?
[#166265842](https://www.pivotaltracker.com/story/show/166265842)
#### Screenshots (if appropriate)
Screenshot 2019-05-31 at 11 41 25 AM
#### Questions:

### CODE REVIEW

**Comment by emp-daisy** [comment]
@PelumiAlesh Please fix your breaking test.

Also, can you help remove this validation from the sign-in route?
Screen Shot 2019-06-04 at 2 26 02 PM

File: server/routes/index.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ import socialRouter from './socialRoute';
 const router = express();
 
 const swaggerApiDoc = yaml.load(`${__dirname}/../docs/hellobooks_api_doc.yaml`);
-
+// /Users/macbookair/Desktop/Projects/HelloBooksLib/server/docs/hellobooks_api_doc.yaml
```
Comment (emp-daisy) [inline server/routes/index.js]: What's this?

Comment (emp-daisy) [reply server/routes/index.js]: @PelumiAlesh

File: server/__test__/mock_data/mock_users.js [hunk 2]
```diff
@@ -11,12 +11,12 @@ const mockUser = {
...
   wrongEmail: {
     email: 'wrongEmail@com',
-    password: 'password'
+    password: 'PassWord123..'
   },
 
   unknownUser: {
     email: 'unknown_person@gmail.com',
-    password: 'password'
+    password: 'PassWord123..'
```
Comment (ocranbillions) [inline server/__test__/mock_data/mock_users.js:19]: is there a reason why you changed all passwords to password123?

File: server/routes/index.js (hunk 1 above)
Comment (PelumiAlesh) [reply server/routes/index.js]: Fixed

File: server/__test__/mock_data/mock_users.js (hunk 2 above)
Comment (PelumiAlesh) [reply server/__test__/mock_data/mock_users.js:19]: I increased password acceptance criteria. It only accepts password that has uppercase letter, lowercase and at least a number

**Review by emp-daisy — APPROVED** [review]

**Review by encodedBicoding — COMMENTED** [review]
coverage is good

**Review by encodedBicoding — APPROVED** [review]

**Comment by PelumiAlesh** [comment]
> coverage is good

Not my fault. I wrote test for all my validations

**Pull request merged**